### PR TITLE
Feature/issue 198

### PR DESCRIPTION
**Pull request by Oleh-Bashtovyi**
## JIRA or Github ticker

* [198](https://github.com/ita-social-projects/VictoryCenter-Back/issues/198)

## Description

Call the required public team members endpoint using Axios methods.

## How it looks

image
image

## Summary of change

The teamPageDataFetch function was updated to use Axios for calling remote endpoints instead of relying on mock data. Tests was implemented for teamPageDataFetch function using jest.

## How to recreate changes

**⚠️ The required GetPublicTeamMembers endpoint in TeamMemberController is not yet merged into the release/1.0.0 on backend, so checkout to "Feature/issue 206" branch on backend repo**

1. Start the `VictoryCenter-Back` server at `https://localhost:5001/`. 
2. In the `VictoryCenter-Client` project, create a `.env` file (if it doesn't exist) and add the following line:  
   `REACT_APP_BASE_API_URL=https://localhost:5001/api/`
3. Start the `VictoryCenter-Client` development server.
4. Open the team page in your browser: `http://localhost:3000/team-page`

## CHECK LIST
- [ ]  New tests was added or existing was modified
- [ ]  Changes has severe impact on users
- [ ]  Changes has medium impact on users
- [ ]  Changes has light impact on users
- [ ]  Test covetage was updated
- [ ]  PR meets all conventions

## Summary by CodeRabbit

* **New Features**
  * The app now fetches team page data from a live API, ensuring up-to-date team information.
  * Introduced a new API service for streamlined HTTP requests.
  * Added centralized API route constants for easier maintenance.

* **Bug Fixes**
  * Improved filtering and validation of team categories and members to ensure only valid data is displayed.

* **Tests**
  * Enhanced and expanded test coverage for team page data fetching, including edge cases and API response handling.
  * Improved type safety in test files for better reliability.

* **Chores**
  * Added the axios library as a new dependency.
  * Updated and extended type definitions for team page data.
  * Removed outdated static mock data for the team page.

### CODE REVIEW

File: src/services/data-fetch/user-pages-data-fetch/team-page-data-fetch/TeamPageDataFetch.ts [hunk 1]
```diff
@@ -1,3 +1,51 @@
...
+    };
+};
+
+export const teamPageDataFetch = async (): Promise<TeamPageData> => {
+    const response = await axiosInstance.get<PublicCategoryWithTeamMembersDto[]>('/TeamMembers/public');
+
+    const teamCategories = response.data
+        .filter(isValidCategory)
+        .map(mapCategoryDtoToTeamCategory)
+        .filter(category => category.members.length > 0);
```
[the remark this answers is not in the document]
Comment (Rominos7) [reply src/services/data-fetch/user-pages-data-fetch/team-page-data-fetch/TeamPageDataFetch.ts:48]: I like more your appoach with removing invalida data
In that way we can ensure that data is usable by app

It is more grasefull way of dealing with invalid data rather then throwing error
+
it can be changed in the future if we need it

File: src/services/data-fetch/user-pages-data-fetch/team-page-data-fetch/TeamPageDataFetch.ts
Comment (VladimirSushinsky) [inline src/services/data-fetch/user-pages-data-fetch/team-page-data-fetch/TeamPageDataFetch.ts:1]: Have you considered using yup for validation?

File: src/services/data-fetch/user-pages-data-fetch/team-page-data-fetch/TeamPageDataFetch.ts [hunk 2]
```diff
@@ -1,3 +1,53 @@
...
+const isValidCategory = (category: PublicCategoryWithTeamMembersDto): boolean => {
+    return Boolean(
+        category?.categoryName?.trim() &&
+        Array.isArray(category.teamMembers) &&
+        category.teamMembers.length > 0
+    );
+};
+
+const isValidTeamMember = (member: PublicTeamMemberDto): boolean => {
+    return Boolean(member?.fullName?.trim());
```
Comment (Rominos7) [inline src/services/data-fetch/user-pages-data-fetch/team-page-data-fetch/TeamPageDataFetch.ts:21]: Just a suggestion (not nessesary to fix)
In order to transform value to boolean you can use [double NOT](double NOT) operator

**Review by ZhmudAnastasiia — APPROVED** [review]
LGTM

File: package.json [hunk 3]
```diff
@@ -13,6 +13,7 @@
         "@testing-library/user-event": "^13.5.0",
```
Comment (MarkBevz50) [inline package.json:13]: Some suggestions
I've heard about the problem with unexpected upadtes of library, so consider locking axios to a specific minor version (e.g., ^1.10.0 → 1.10.0) to avoid such issues.

File: src/services/data-fetch/user-pages-data-fetch/team-page-data-fetch/TeamPageDataFetch.ts [hunk 4]
```diff
@@ -1,3 +1,48 @@
-import { TeamPageMock } from '../../../../utils/mock-data/user-pages/team-page/TeamPage';
```
Comment (MarkBevz50) [inline src/services/data-fetch/user-pages-data-fetch/team-page-data-fetch/TeamPageDataFetch.ts:1]: Some suggestions:
Add error handling with try/catch for the axios.get call.
Im not 100% sure, but consider caching the response to avoid redundant API calls.

**Review by MarkBevz50 — APPROVED** [review]
There are some insignificant issues, but overall, it's ready to merge.

File: src/services/data-fetch/user-pages-data-fetch/team-page-data-fetch/TeamPageDataFetch.ts (hunk 4 above)
Comment (Oleh-Bashtovyi) [reply src/services/data-fetch/user-pages-data-fetch/team-page-data-fetch/TeamPageDataFetch.ts:1]: It is already handling on frontend side
